clang/lib/Driver/Driver.cpp

```diff
@@ -3787,6 +3785,53 @@ class OffloadingActionBuilder final {
       SYCLDeviceActions.clear();
     }
 
+    void addSYCLDeviceLibs(const ToolChain *TC, ActionList &DeviceLinkObjects,
+                           bool isSpirvAOT, bool isMSVCEnv) {
+      enum SYCLDeviceLibType {
+        sycl_devicelib_wrapper,
+        sycl_devicelib_fallback
+      };
+      StringRef LibLoc, LibSysUtils;
+      if (isMSVCEnv) {
+        LibLoc = Args.MakeArgString(TC->getDriver().Dir + "/../bin");
+        LibSysUtils = "libsycl-msvc";
+      } else {
+        LibLoc = Args.MakeArgString(TC->getDriver().Dir + "/../lib");
+        LibSysUtils = "libsycl-glibc";
+      }
+      SmallVector<StringRef, 4> sycl_device_wrapper_libs = {
+          LibSysUtils, "libsycl-complex", "libsycl-complex-fp64",
+          "libsycl-cmath", "libsycl-cmath-fp64"};
+      // For AOT compilation, we need to link sycl_device_fallback_libs as
+      // default too.
+      SmallVector<StringRef, 4> sycl_device_fallback_libs = {
+          "libsycl-fallback-cassert", "libsycl-fallback-complex",
+          "libsycl-fallback-complex-fp64", "libsycl-fallback-cmath",
+          "libsycl-fallback-cmath-fp64"};
+      auto addInputs = [&](SYCLDeviceLibType t) {
+        auto sycl_libs = (t == sycl_devicelib_wrapper)
+                             ? sycl_device_wrapper_libs
+                             : sycl_device_fallback_libs;
+        for (const StringRef &Lib : sycl_libs) {
+          SmallString<128> LibName(LibLoc);
+          llvm::sys::path::append(LibName, Lib);
+          llvm::sys::path::replace_extension(LibName, ".o");
```

Comment:
`.o` is not a standard object name on Windows (typically it is `.obj`).  Probably can be discussed/addressed separately, but having the naming be inconsistent with expectations is confusing.

Reply:
Hi, @mdtoguchi 
As we discussed above, current device libraries use ".o" for both Linux and Windows. I agree that ".obj" should be used for Windows and I think we can address separately in another PR.
Thank you very much.